src/odemis/acq/test/stream_test.py

```diff
@@ -2200,6 +2201,356 @@ def test_acq_cl_leech(self):
+        self.assertNotEqual(streaks.detMCPGain.value, 0)
+        # checks that HW VA and GUI VA are equal when stream active
+        self.assertEqual(self.streak_unit.MCPGain.value, streaks.detMCPGain.value)
+
+    def test_streak_acq(self):
+        """Test acquisition with streak camera"""
+
+        # Create the stream
+        sems = stream.SEMStream("test sem", self.sed, self.sed.data, self.ebeam)
+        # test with streak camera
+        streaks = stream.TemporalSpectrumSettingsStream("test streak cam", self.streak_ccd, self.streak_ccd.data,
+                                                        self.ebeam, self.streak_unit, self.streak_delay,
+                                                        streak_unit_vas={"timeRange", "MCPGain", "streakMode"})
+
+        stss = stream.SEMTemporalSpectrumMDStream("test sem-tempSpectrum", [sems, streaks])
+
+        streaks.detStreakMode.value = True
+
+        self.streak_ccd.exposureTime.value = 0.01  # 10ms
+        # # TODO use fixed repetition value -> set ROI?
+        streaks.repetition.value = (10, 5)
+        num_tempSpec = numpy.prod(streaks.repetition.value)  # number of expected tempSpec images
+        exp_pos, exp_pxs, exp_res = self._roiToPhys(streaks)
+
+        # Start acquisition
+        # estimated acquisition time should be accurate with less than 50% margin + 1 extra second
+        timeout = 10 + 1.5 * stss.estimateAcquisitionTime()  # TODO check with real HW! timedout problems
+        start = time.time()
+        f = stss.acquire()  # calls acquire method in MultiDetectorStream in sync.py
+
+        # stss.raw: array containing as first entry the sem scan image for the scanning positions,
+        # the second array are temporal spectrum images
+        # data: array should contain same images as stss.raw
+
+        # wait until it's over
+        data = f.result(timeout)
+        dur = time.time() - start
+        logging.debug("Acquisition took %g s", dur)
+        self.assertTrue(f.done())
+
+        # check if number of images in the received data (sem image + tempSpec images) is the same as
+        # number of images stored in raw
+        self.assertEqual(len(data), len(stss.raw))
+
+        # check that sem data array has same shape as expected for the scanning positions of ebeam
+        sem_da = stss.raw[0]  # sem data array for scanning positions
+        self.assertEqual(sem_da.shape, exp_res[::-1])
+
+        # check that sem data array has same shape as expected for the scanning positions of ebeam
+        tempSpec_da = stss.raw[1]  # temporal spectrum data array
+        shape = tempSpec_da.shape
+        self.assertEqual(shape[3] * shape[4], num_tempSpec)
+        # len of shape should be 5: CTZXY
+        self.assertEqual(len(shape), 5)
+
+        # check if metadata is correctly stored
+        md = tempSpec_da.metadata
+        self.assertIn(model.MD_STREAK_TIMERANGE, md)
+        self.assertIn(model.MD_STREAK_MCPGain, md)
```

Comment:
`MD_STREAK_MCPGAIN` (all uppercase). 
Please run the test cases again, to check they still pass with the latest version of the (simulated) driver.